### PR TITLE
OpenEXR linking fix for Windows when using an external library that is dynamic linked

### PR DESCRIPTION
**Pull request by boberfly**
This helps when building Appleseed against Gaffer Windows

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -123,6 +123,7 @@ option (USE_STATIC_BOOST                    "Use static Boost libraries"
 option (USE_STATIC_OIIO                     "Use static OpenImageIO libraries"                      ON)
 option (USE_STATIC_OCIO                     "Use static OpenColorIO libraries"                      ON)
 option (USE_STATIC_OSL                      "Use static OpenShadingLanguage libraries"              ON)
+option (USE_STATIC_EXR                      "Use static OpenEXR libraries"                          ON)
```
Comment (dictoon) [inline CMakeLists.txt]: Can you move this up to follow the same ordering as the `USE_EXTERNAL_*` options?

File: CMakeLists.txt [hunk 2]
```diff
@@ -226,6 +227,9 @@ endif ()
 if (USE_EXTERNAL_EXR)
     find_package (Imath REQUIRED)
     find_package (OpenEXR REQUIRED)
+    if (NOT USE_STATIC_EXR AND MSVC)
+        add_definitions(-DOPENEXR_DLL)
```
Comment (dictoon) [inline CMakeLists.txt]: Add a space after `add_definitions`.

**Comment by boberfly** [comment]
Shall I move the check to win-vs.txt also?

**Comment by dictoon** [comment]
> Shall I move the check to win-vs.txt also?

If possible, yes. Thanks!

File: cmake/config/win-vs.txt [hunk 3]
```diff
@@ -69,6 +69,13 @@ set (preprocessor_definitions_release
     _SECURE_SCL=0
 )
 
+if (NOT USE_STATIC_EXR)
+    set (preprocessor_definitions_common
+    ${preprocessor_definitions_common}
```
Comment (dictoon) [inline cmake/config/win-vs.txt]: Fix indentation.

Comment (boberfly) [reply cmake/config/win-vs.txt]: Ah my bad, fixed

**Pull request merged**